src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -187,6 +180,24 @@ def register_options(cls, register):
 
   def register_extra_products_from_contexts(self, targets, compile_contexts):
     super(RscCompile, self).register_extra_products_from_contexts(targets, compile_contexts)
+
+    def cp_entry_for(maybe_file):
+      # Capture a snapshot for the file if it exists
+      if os.path.exists(maybe_file):
+        def capture_or_read_digest(scheduler, filename):
+          digest = load_digest(os.path.dirname(filename))
+          if not digest:
+            root = PathGlobsAndRoot(
+              PathGlobs(
+                (fast_relpath_optional(maybe_file, get_buildroot()),)),
+              text_type(get_buildroot()))
+            digest = scheduler.capture_snapshots((root,))[0].directory_digest
```

Comment:
This probably comes from me misreading something then. I think I was thinking here that classpath entries need to be 1:1 with their digests. Also, I added the capture here before I added the dropping and loading of digests from files, and looking at it now, it may not be necessary.